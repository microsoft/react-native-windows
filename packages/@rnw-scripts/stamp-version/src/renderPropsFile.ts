/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import fs from '@react-native-windows/fs';
import mustache from 'mustache';
import path from 'path';
import semver from 'semver';

/**
 * Render PackageVersion.g.props
 */
export default async function renderPropsFile(
  version: string,
): Promise<string> {
  const templatePath = path.join(
    __dirname,
    '..',
    'template',
    'PackageVersion.g.props.hbs',
  );
  const templateContent = await fs.readFile(templatePath);

  return mustache.render(templateContent.toString(), {
    version,
    major: semver.major(version),
    minor: semver.minor(version),
    patch: semver.patch(version),
    canary: semver.prerelease(version)?.[0] === 'canary',
  });
}
