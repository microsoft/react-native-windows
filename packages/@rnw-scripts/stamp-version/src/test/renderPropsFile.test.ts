/**
 * Copyright (c) Microsoft Corporation.
 * Licensed under the MIT License.
 *
 * @format
 */

import renderPropsFile from '../renderPropsFile';

test('0.0.0-canary.100', async () => {
  expect(await renderPropsFile('0.0.0-canary.100')).toMatchSnapshot();
});

test('0.67.0-preview.1', async () => {
  expect(await renderPropsFile('0.67.0-preview.1')).toMatchSnapshot();
});

test('0.66.5', async () => {
  expect(await renderPropsFile('0.66.5')).toMatchSnapshot();
});
