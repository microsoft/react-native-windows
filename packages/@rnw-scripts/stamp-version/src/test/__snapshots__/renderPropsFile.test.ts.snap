// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`0.0.0-canary.100 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<!-- 
  Copyright (c) Microsoft Corporation.
  Licensed under the MIT License.

  Generated by @rnw-scripts/stamp-version

  This file is stamped with the version of the published react-native-windows
  npm package.
-->
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <ReactNativeWindowsVersion>0.0.0-canary.100</ReactNativeWindowsVersion>
    <ReactNativeWindowsMajor>0</ReactNativeWindowsMajor>
    <ReactNativeWindowsMinor>0</ReactNativeWindowsMinor>
    <ReactNativeWindowsPatch>0</ReactNativeWindowsPatch>
    <ReactNativeWindowsCanary>true</ReactNativeWindowsCanary>
  </PropertyGroup>
</Project>"
`;

exports[`0.66.5 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<!-- 
  Copyright (c) Microsoft Corporation.
  Licensed under the MIT License.

  Generated by @rnw-scripts/stamp-version

  This file is stamped with the version of the published react-native-windows
  npm package.
-->
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <ReactNativeWindowsVersion>0.66.5</ReactNativeWindowsVersion>
    <ReactNativeWindowsMajor>0</ReactNativeWindowsMajor>
    <ReactNativeWindowsMinor>66</ReactNativeWindowsMinor>
    <ReactNativeWindowsPatch>5</ReactNativeWindowsPatch>
    <ReactNativeWindowsCanary>false</ReactNativeWindowsCanary>
  </PropertyGroup>
</Project>"
`;

exports[`0.67.0-preview.1 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<!-- 
  Copyright (c) Microsoft Corporation.
  Licensed under the MIT License.

  Generated by @rnw-scripts/stamp-version

  This file is stamped with the version of the published react-native-windows
  npm package.
-->
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <PropertyGroup>
    <ReactNativeWindowsVersion>0.67.0-preview.1</ReactNativeWindowsVersion>
    <ReactNativeWindowsMajor>0</ReactNativeWindowsMajor>
    <ReactNativeWindowsMinor>67</ReactNativeWindowsMinor>
    <ReactNativeWindowsPatch>0</ReactNativeWindowsPatch>
    <ReactNativeWindowsCanary>false</ReactNativeWindowsCanary>
  </PropertyGroup>
</Project>"
`;
